crmd/notify.c

```diff
@@ -20,120 +20,501 @@
+                nano_digits = (nano_digits > 6)?6:nano_digits;
+                nano_digits = (nano_digits < 0)?0:nano_digits;
+                sprintf(&nanofmt_s[1], ".%ds", nano_digits);               
+            } else {
+                if (format[fmt_pos+fmt_len] != 0) {
+                    continue;
+                }
+                fmt_pos = scanned_pos; /* print till end */
+            }
+        } else {
+            scanned_pos = strlen(format);
+            fmt_pos = scanned_pos; /* print till end */
+        }
+        tmp_fmt_s = strndup(&format[printed_pos], fmt_pos - printed_pos);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+        date_len += strftime(&date_s[date_len], max-date_len, tmp_fmt_s, &tm);
+#pragma GCC diagnostic pop
+        printed_pos = scanned_pos;
+        free(tmp_fmt_s);
+        if (nano_digits) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+            date_len += snprintf(&date_s[date_len], max-date_len, 
+                                 nanofmt_s, nano_s);
+#pragma GCC diagnostic pop 
+            nano_digits = 0;
+        }
+    }
+
+    return (date_len == 0)?NULL:strdup(date_s);
+}
+
+/*
+ * end of possibly generic time-handling stuff
+ */
+
+
+/*
+ * syncronize local data with cib
+ */
+
+static void free_notify_list_entry(notify_entry_t *entry)
+{
+    free(entry->id);
+    free(entry->path);
+    free(entry->tstamp_format);
+    free(entry->info);
+    free(entry->recipient);
+    free(entry);
+}
+
+static void free_notify_list()
+{
+    if (notify_list) {
+        g_list_free_full(notify_list, (GDestroyNotify) free_notify_list_entry);
+        notify_list = NULL;
+    }
+}
+
+static void add_dup_notify_list_entry(notify_entry_t *entry)
+{
+    notify_entry_t *new_entry = 
+        (notify_entry_t *) calloc(1, sizeof(notify_entry_t));
+
+    CRM_ASSERT(new_entry);
+    *new_entry = (notify_entry_t) {
+       .id = strdup(entry->id),
+       .path = strdup(entry->path),
+       .timeout = entry->timeout,
+       .tstamp_format = entry->tstamp_format?strdup(entry->tstamp_format):NULL,
+       .info = entry->info?strdup(entry->info):NULL,
+       .recipient = entry->recipient?strdup(entry->recipient):NULL
+    };
+    notify_list = g_list_append(notify_list, new_entry);
+}
+
+void
+notifications_query_callback(xmlNode * msg, int call_id, int rc, 
+                             xmlNode * output, void *user_data)
+{
+    xmlNode *notify = output;
+    notify_entry_t entry;
+
+    free_notify_list();
+
+    if (notify_script) {
+        entry = (notify_entry_t) {
+            .id = notify_script,
+            .path = notify_script,
+            .timeout = CRMD_NOTIFY_DEFAULT_TIMEOUT_MS,
+            .recipient = notify_target
+        };
+        add_dup_notify_list_entry(&entry);
+        crm_notice("Legacy Notifications enabled");
+    } else {
+        crm_notice("Legacy Notifications disabled");
+    }
```

Comment:
Good comment. I already fooled myself needing a couple of thoughts why my sequence 
counter was counting up in steps of 2 ;-)
So I go for existence of entries in the new section disabling the legacy feature and if
the legacy-properties are present as well it will result in a warning. The notice that 
the legacy-feature is being used should probably stay.